tools/buildings_to_dsm.py

```diff
@@ -277,56 +304,61 @@ def main(args):
-            dsmElevation = elevation
-        else:
-            # elevation has nans in places other than buildings
-            dsmElevation = numpy.fmax(dtmRaster, elevation)
-        dsm.GetRasterBand(1).WriteArray(dsmElevation)
-        if nodata:
-            dsm.GetRasterBand(1).SetNoDataValue(nodata)
+        try:
+
+            ren.ResetCamera()
+            camera = ren.GetActiveCamera()
+            camera.ParallelProjectionOn()
+            camera.SetParallelScale((dtmBounds[3] - dtmBounds[2])/2)
+            distance = camera.GetDistance()
+            focalPoint = [(dtmBounds[0] + dtmBounds[1]) * 0.5,
+                          (dtmBounds[3] + dtmBounds[2]) * 0.5,
+                          (buildingsScalarRange[0] + buildingsScalarRange[1]) * 0.5]
+            position = [focalPoint[0], focalPoint[1], focalPoint[2] + distance]
+            camera.SetFocalPoint(focalPoint)
+            camera.SetPosition(position)
+
+            valuePass = vtk.vtkValuePass()
+            valuePass.SetRenderingMode(vtk.vtkValuePass.FLOATING_POINT)
+            # use the default scalar for point data
+            valuePass.SetInputComponentToProcess(0)
+            valuePass.SetInputArrayToProcess(vtk.VTK_SCALAR_MODE_USE_POINT_FIELD_DATA,
+                                             arrayName)
+            passes = vtk.vtkRenderPassCollection()
+            passes.AddItem(valuePass)
+            sequence = vtk.vtkSequencePass()
+            sequence.SetPasses(passes)
+            cameraPass = vtk.vtkCameraPass()
+            cameraPass.SetDelegatePass(sequence)
+            ren.SetPass(cameraPass)
+            # We have to render the points first, otherwise we get a segfault.
+            renWin.Render()
+            valuePass.SetInputArrayToProcess(vtk.VTK_SCALAR_MODE_USE_CELL_FIELD_DATA, arrayName)
+            renWin.Render()
+            elevationFlatVtk = valuePass.GetFloatImageDataArray(ren)
+            valuePass.ReleaseGraphicsResources(renWin)
+
+            print("Writing the DSM ...")
+            elevationFlat = numpy_support.vtk_to_numpy(elevationFlatVtk)
+            # VTK X,Y corresponds to numpy cols,rows. VTK stores arrays
+            # in Fortran order.
+            elevationTranspose = numpy.reshape(
+                elevationFlat, [dtm.RasterXSize, dtm.RasterYSize], "F")
+            # changes from cols, rows to rows,cols.
+            elevation = numpy.transpose(elevationTranspose)
+            # numpy rows increase as you go down, Y for VTK images increases as you go up
+            elevation = numpy.flip(elevation, 0)
+            if args.buildings_only:
+                dsmElevation = elevation
+            else:
+                # elevation has nans in places other than buildings
+                dsmElevation = numpy.fmax(dtmRaster, elevation)
+            dsm.GetRasterBand(1).WriteArray(dsmElevation)
+            if nodata:
+                dsm.GetRasterBand(1).SetNoDataValue(nodata)
+
+        except:
+            pass
 
```

Comment:
Who throws the exception? Why pass and not print the error?

Reply:
I wonder if this was just added for debugging before the offscreen rendering was fixed.  Maybe we can ignore this change.  It doesn't seem like a good idea to catch all exceptions and then pass without notification.